sklearn/metrics/_pairwise_distances_reduction.pyx

```diff
@@ -1088,3 +1190,631 @@ cdef class FastEuclideanPairwiseDistancesArgKmin(PairwiseDistancesArgKmin):
+            respectively on chunks of X and chunks of Y.
+            Strategies differs on which loop (outer or inner) is made to run
+            in parallel with the Cython `prange` construct:
+
+              - 'parallel_on_X' dispatches chunks of X uniformly on threads.
+              Each thread then iterates on all the chunks of Y. This strategy is
+              embarrassingly parallel and comes with no datastructures synchronisation.
+
+              - 'parallel_on_Y' dispatches chunks of Y uniformly on threads.
+              Each thread processes all the chunks of X in turn. This strategy is
+              a sequence of embarrassingly parallel subtasks (the inner loop on Y
+              chunks) with intermediate datastructures synchronisation at each
+              iteration of the sequential outer loop on X chunks.
+
+              - 'auto' relies on a simple heuristic to choose between
+              'parallel_on_X' and 'parallel_on_Y': when `X.shape[0]` is large enough,
+              'parallel_on_X' is usually the most efficient strategy. When `X.shape[0]`
+              is small but `Y.shape[0]` is large, 'parallel_on_Y' brings more opportunity
+              for parallelism and is therefore more efficient despite the synchronization
+              step at each iteration of the outer loop on chunks of `X`.
+
+              - None (default) looks-up in scikit-learn configuration for
+              `pairwise_dist_parallel_strategy`, and use 'auto' if it is not set.
+
+        return_distance : boolean, default=False
+            Return distances between each X vector and its neighbors if set to True.
+
+        sort_results : boolean, default=False
+            Sort results with respect to distances between each X vector and its
+            neighbors if set to True.
+
+        Returns
+        -------
+            If return_distance=False:
+              - neighbors_indices : ndarray of n_samples_X ndarray
+                Indices of the neighbors for each vector in X.
+
+            If return_distance=True:
+              - neighbors_indices : ndarray of n_samples_X ndarray
+                Indices of the neighbors for each vector in X.
+              - neighbors_distances : ndarray of n_samples_X ndarray
+                Distances to the neighbors for each vector in X.
+
+        Notes
+        -----
+            This public classmethod is responsible for introspecting the arguments
+            values to dispatch to the private
+            :meth:`PairwiseDistancesRadiusNeighborhood._compute` instance method of
+            the most appropriate :class:`PairwiseDistancesRadiusNeighborhood`
+            concrete implementation.
+
+            All temporarily allocated datastructures necessary for the concrete
+            implementation are therefore freed when this classmethod returns.
+
+            This allows entirely decoupling the interface entirely from the
+            implementation details whilst maintaining RAII.
```

Comment:
```suggestion
        This public classmethod is responsible for introspecting the arguments
        values to dispatch to the private
        :meth:`PairwiseDistancesRadiusNeighborhood._compute` instance method of
        the most appropriate :class:`PairwiseDistancesRadiusNeighborhood`
        concrete implementation.

        All temporarily allocated datastructures necessary for the concrete
        implementation are therefore freed when this classmethod returns.

        This allows entirely decoupling the interface entirely from the
        implementation details whilst maintaining RAII.
```

feel free to rewrap but otherwise it's fine like this.